sodium/build.rs

```diff
@@ -0,0 +1,31 @@
+use libc;
+use std::io::Error;
+use std::ptr::null_mut;
+
+// Example custom build script.
+fn main() {
+    // 1. check for memfd_secret support
+    let fd  = unsafe {
+        libc::syscall(libc::SYS_memfd_secret, 0) as i32
+    };
+    if fd != -1 || if let Some(os_err) = Error::last_os_error().raw_os_error() { os_err != 38 } else { true } {
+        // with memfd_secret support
+        println!("cargo:rustc-cfg=with_memfd_secret");
+    }
+    if fd != -1 {
+        unsafe {
+            libc::close(fd);
+        }
+    }
```

Comment:
This is a leaky abstraction, that only produces correct results when `buildPlatform == hostPlatform`. This is not good, Rust has quite good cross-compilation support, and once we get rid of our C dependencies it should be trivial to cross compile Rosenpass. If I understand this code correctly, it breaks that, because it uses syscalls on the buildPlatform to detect which features may be available on the hostPlatform.

Reply:
If detecting syscalls is not possible at compile time, we should do it at runtime!